### PR TITLE
docs: Resources section improvements

### PR DESCRIPTION
**Pull request by andrii-bodnar**
# Description

🖼️ [Deployment Preview](https://js-lingui-git-docs-resources-crowdin.vercel.app/)

## Types of changes

[//]: # (What types of changes does your code introduce to Lingui?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Examples update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/lingui/js-lingui/blob/main/CONTRIBUTING.md) and [CODE_OF_CONDUCT](https://github.com/lingui/js-lingui/blob/main/CODE_OF_CONDUCT.md) docs
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation (if appropriate)

### CODE REVIEW

File: website/docs/misc/i18next.md [hunk 1]
```diff
@@ -45,48 +51,72 @@ module.exports = {
...
 
+If you prefer to use explicit IDs for your messages, you can utilize the `msg` macro:
+
+```js
+import { msg } from "@lingui/core/macro";
+
+msg({ id: "msg.greeting", message: `Hello World` });
+```
+
+Read more about [Explicit vs Generated Message IDs](/docs/guides/explicit-vs-generated-ids.md).
```
Comment (timofei-iatsenko) [inline website/docs/misc/i18next.md:71]: It also possible tow write explicit id with the `t` macro, 

```js
import { t } from "@lingui/core/macro";

document.getElementById("output").innerHTML = t({ id: "msg.greeting", message: `Hello World` });
```

The difference between `msg` and `t` is first one returns `MessageDescriptor`, second one wraps this message descriptor into global `i18n._` method, so returns a translated string directly. The signature of these to macros is almost the same.

File: website/docs/misc/i18next.md [hunk 2]
```diff
@@ -45,48 +51,72 @@ module.exports = {
...
 Lingui sample:
 
 ```js
-i18n._({ id: "msg.name", message: "My name is {name}", values: { name: "Tom" } });
+import { msg, t } from "@lingui/core/macro";
+
+const name = "Tom";
+
+t`My name is ${name}`;
+msg({ id: "msg.name", message: `My name is ${name}` });
```
Comment (timofei-iatsenko) [inline website/docs/misc/i18next.md:94]: As commented earlier, more correct analog would be: 

```js
t`My name is ${name}`;
t({ id: "msg.name", message: `My name is ${name}` });
```

File: website/docs/misc/i18next.md [hunk 3]
```diff
@@ -185,37 +218,78 @@ i18next handles plurals differently. It requires a separate key to be defined fo
...
+Lingui sample:
+
+```js
+import { msg } from "@lingui/core/macro";
+
+msg({
+  message: "Right",
+  context: "direction",
+});
+```
```
Comment (timofei-iatsenko) [inline website/docs/misc/i18next.md:249]: Would it be usefull to add a `po` example here to show what was written in the description above: 

> Lingui provides an automatic additional "context" by including in the `.po` file the locations where each message is used - i18next can't do this from its plain JSON files.

File: website/docs/misc/i18next.md [hunk 4]
```diff
@@ -185,37 +218,78 @@ i18next handles plurals differently. It requires a separate key to be defined fo
...
+Lingui sample:
+
+```jsx
+import { Trans } from "@lingui/react/macro";
+
+const HelloWorld = () => {
+  return <Trans>Hello World!</Trans>;
+};
+```
+
```
Comment (timofei-iatsenko) [inline website/docs/misc/i18next.md:274]: I think the main diffrence between this two, and key selling point of using macro over runtime jsx elements is ability to use child components in the string and don't split message into chunks. I think it's worth to highlight it here.

```jsx
import { Trans } from "react-i18next";

const HelloWorld = () => {
  return <Trans i18nKey="link.check">Check</Trans>{' '}<a href="#"><Trans i18nKey="link.this">this!</Trans</a>;
};
```

Lingui sample:

```jsx
import { Trans } from "@lingui/react/macro";

const HelloWorld = () => {
  return <Trans>Check <a href="#">this!</a></Trans>;
};
```

**Comment by timofei-iatsenko** [comment]
Reviewed the i18next page, will come back to the rest a bit later.

File: website/docs/misc/i18next.md (hunk 4 above)
Comment (andrii-bodnar) [reply website/docs/misc/i18next.md:274]: Based on the documentation (https://react.i18next.com/latest/trans-component), it doesn't seem like it should be concatenated that way like in your example

Comment (timofei-iatsenko) [reply website/docs/misc/i18next.md:287]: Wow, you are right, I've believed that this is not possible with a plain JSX. Hence created this example.

Now I'm extremely interested how do they do that.

**Pull request merged**